Merge/foundation release/1.10.2 resolved

meowsbits opened the pull request:
Stacked on https://github.com/etclabscore/core-geth/pull/348.

---

See [related Tracking Issue](https://github.com/etclabscore/core-geth/issues/365) for TODOs and review status.

karalabe committed:
params: begin v1.10.2 release cycle

MariusVanDerWijden committed:
core/types: reduce allocations in GasPriceCmp (#22456)

holiman committed:
les: fix errors in metric namespace (#22459)

* les: add trailing slash to metric namespace

* les: omit '.' in metric namespace

rjl493456442 committed:
cmd: extend dumpgenesis to support network flags on the cmd (#22406)

fjl committed:
p2p/enr: fix decoding of incomplete lists (#22484)

Given a list of less than two elements DecodeRLP returned rlp.EOL,
leading to issues in outer decoders.

michael1011 committed:
cmd/geth: add ancient datadir flag to snapshot subcommands (#22486)

holiman committed:
cmd/devp2p: better testcase failure output for ethtests (#22482)

holiman committed:
eth, les: properly init statedb accesslist during tracing (#22480)

* eth/state, les/state: properly init statedb accesslist when tracing, fixes #22475

* eth: review comments

* eth/tracers: fix compilation err

* eth/tracers: apply @karalabe's suggested fix

holiman committed:
cmd/geth, eth/downloader: remove copydb command (#22501)

* cmd/geth: remove copydb command

* eth/downloader: remove fakepeer

holiman committed:
tests/fuzzers: fix goroutine leak in les fuzzer (#22455)

The oss-fuzz fuzzer has been reporting some failing testcases for les. They're all spurious, and cannot reliably be reproduced. However, running them showed that there was a goroutine leak: the tests created a lot of new clients, which started an exec queue that was never torn down.

This PR fixes the goroutine leak, and also a log message which was erroneously formatted.

lightclient committed:
core/types: improve comments in new EIP-2718 code (#22402)

Responding to these comments:

    https://github.com/ethereum/go-ethereum/pull/21502/files#r579010962
    https://github.com/ethereum/go-ethereum/pull/21502/files#r579021565
    https://github.com/ethereum/go-ethereum/pull/21502/files#r579023510
    https://github.com/ethereum/go-ethereum/pull/21502/files#r578983734

ligi committed:
cmd/clef: docs - link to ethereum org repo (#22400)

ligi committed:
cmd/clef (docs): fix image background (#22399)

Flatten the image so we do not have dark text on dark background

holiman committed:
core/rawdb: fix transaction indexing/unindexing hashing error (#22457)

* core/rawdb: more verbose error logs + better hashing

* core/rawdb: add failing testcase

* core/rawdb: properly hash transactions while indexing/unindexing

* core/rawdb: exit on error + better log msg

zsfelfoldi committed:
les: fix UDP connection query (#22451)

This PR fixes multiple issues with the UDP connection pre-negotiation feature:

- the enable condition was wrong (it checked the existence of the DiscV5 struct where it wasn't initialized yet, disabling the feature even if discv5 was enabled)
- the server pool queried already connected nodes when the discovery iterators returned them again
- servers responded positively before they were synced and really willing to accept connections

Metrics are also added on the server side that count the positive and negative replies to served connection queries.

zsfelfoldi committed:
les: fix UDP connection query (#22451)

This PR fixes multiple issues with the UDP connection pre-negotiation feature:
- the enable condition was wrong (it checked the existence of the DiscV5 struct where it wasn't initialized yet, disabling the feature even if discv5 was enabled)
- the server pool queried already connected nodes when the discovery iterators returned them again
- servers responded positively before they were synced and really willing to accept connections

Metrics are also added on the server side that count the positive and negative replies to served connection queries.

zsfelfoldi committed:
les: allow either full enode strings or raw hex ids in the API (#22423)

holiman committed:
eth/protocols/snap, eth/downloader: don't use bloom filter in snap sync

r1cs committed:
eth/protocols/snap: fix typo (#22530)

renaynay committed:
cmd/devp2p/internal/ethtest: return request ID in BlockHeaders response (#22508)

This PR fixes an issue with the eth66 test suite where, during a readAndServe when
the test is manually responding to GetBlockHeader requests, it now responds
with a BlockHeaders eth66 packet that includes the inbound request ID.

meowsbits committed:
ethclient: fix error handling for header test (#22514)

The wantErr field was disused, and the error returned by HeaderByNumber
was not properly tested.

This simplifies the error checking using errors.Is and asserts that getting
an expected missing header returns ethereum.NotFound. Also adds a nil
check condition for header.Number before using big.Int's Sign method.

reds committed:
accounts/abi/bind: add NoSend transact option (#22446)

This adds a new option to avoid sending the transaction which is
created by calling a bound contract method.

ucwong committed:
go.mod: upgrade goleveldb to commit 64b5b1c (#22436)

This pulls in a fix for a corruption issue when the process crashes
while a new manifest file is being added.

ucwong committed:
go.mod: upgrade goupnp to commit 0ca76305 (#22479)

This pulls in a fix to skip the broadcast on interfaces which are down.

jacksoom committed:
consensus/ethash: remove unnecessary variable definition (#22512)

qhenkart committed:
cmd/devp2p: use AWS-SDK v2 (#22360)

This updates the DNS deployer to use AWS SDK v2. Migration is relatively
seamless, although there were two locations that required a slightly
different approach to achieve the same results. In particular, waiting for
DNS change propagation is very different with SDK v2. 

This change also optimizes DNS updates by publishing all changes before
waiting for propagation.

holiman committed:
p2p/dnsdisc: fix flaw in dns size calculation (#22533)

This fixes the calculation of the tree branch factor. With the new
formula, we now creat at most 13 children instead of 30, ensuring
the TXT record size will be below 370 bytes.

s1na committed:
core: fix potential race in chainIndexerTest (#22346)

renaynay committed:
cmd/devp2p/internal/ethtest: skip eth/66 tests when v66 not supported (#22460)

fjl committed:
cmd/devp2p: add flag for AWS region (#22537)

holiman committed:
cmd/devp2p: fix error in updating the cursor when collecting records for route53 (#22538)

This PR fixes a regression introduced in #22360, when we updated to the v2 of the AWS sdk, which causes current crawler to just get the same first 100 results over and over, and get stuck in a loop.

renaynay committed:
cmd/devp2p: add old block announcement test to eth test suite (#22474)

Add old block announcement test to eth test suite, checks to make sure old block announcement isn't propagated

tobias-hildebrandt committed:
cmd/utils: fix compilation issue on openbsd (#22511)

derekchiang committed:
core: fix method comment for `txpool.requestReset` (#22543)

MrChico committed:
accounts: eip-712 signing for ledger (#22378)

* accounts: eip-712 signing for ledger

* address review comments

rjl493456442 committed:
all: add read-only option to database (#22407)

* all: add read-only option to database

* all: fixes tests

* cmd/geth: migrate flags

* cmd/geth: fix the compact

* cmd/geth: fix the format

* cmd/geth: fix log

* cmd: add chain-readonly

* core: add readonly notion to freezer

* core/rawdb: add log

* core/rawdb: fix freezer close

* cmd: fix

* cmd, core: construct db

* core: update tests

s1na committed:
cmd/geth: check block range against chain head in export cmd (#22387)

Check the input parameters against the actual head block, exit on error

karalabe committed:
core/state/snapshot: fix panic on missing parent

karalabe committed:
Merge pull request #22551 from karalabe/snapshot-parent-get-panic

core/state/snapshot: fix panic on missing parent

karalabe committed:
Merge pull request #22513 from holiman/no_snap_bloom

eth/protocols/snap, eth/downloader: don't use bloom filter in snap sync

meowsbits committed:
internal/web3ext, node: migrate node admin API (Start|Stop)RPC->HTTP (#22461)

* internal/web3ext,node: migrate node admin API (Start|Stop)RPC->HTTP

Corresponding CLI flags --rpc have been moved to --http.

This moves the admin module HTTP RPC start/stop
methods to an equivalent namespace.

Rel https://github.com/ethereum/go-ethereum/pull/22263

Date: 2021-03-08 08:00:11-06:00
Signed-off-by: meows <b5c6@protonmail.com>

* internal/web3ext: fix startRPC/HTTP param count (4->5)

Date: 2021-03-16 06:13:23-05:00
Signed-off-by: meows <b5c6@protonmail.com>

fjl committed:
cmd/devp2p: skip ENR field tails properly in nodeset filter (#22565)

In Geth v1.10, we changed the structure of the "les" ENR entry. As a result, the DHT crawler that creates the DNS lists
no longer recognizes the les nodes, which is fixed in this commit. 

* cmd/devp2p: skip ENR field tails properly in nodeset filter
* cmd/devp2p: fix tail decoder for snap as well
* les: fix tail decoding in "eth" ENR entry

terasum committed:
p2p: fix minor typo and remove fd parameter in checkInboundConn (#22547)

fjl committed:
p2p/dnsdisc: rate limit resolving before checking cache (#22566)

This makes the rate limit apply regardless of whether the node is
already cached.

rjl493456442 committed:
eth/protocols/snap: fix the flaws in the snap sync (#22553)

* eth/protocols/snap: fix snap sync

* eth/protocols/snap: fix tests

* eth: fix tiny

* eth: update tests

* eth: update tests

* core/state/snapshot: testcase for #22534

* eth/protocols/snap: fix boundary loss on full-but-proven range

* core/state/snapshot: lintfix

* eth: address comment

* eth: fix handler

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Péter Szilágyi <peterke@gmail.com>

holiman committed:
eth/tracers, core: use scopecontext in tracers, provide statedb in capturestart (#22333)

Fixes the CaptureStart api to include the EVM, thus being able to set the statedb early on. This pr also exposes the struct we used internally in the interpreter to encapsulate the contract, mem, stack, rstack, so we pass it as a single struct to the tracer, and removes the error returns on the capture methods.

holiman committed:
core: fix condition on header verification

fjl committed:
cmd/devp2p: fix comparison of TXT record value (#22572)

* cmd/devp2p: fix comparison of TXT record value

The AWS API returns quoted DNS strings, so we must encode the new value
before comparing it against the existing record content.

* cmd/devp2p: add test

* cmd/devp2p: fix typo and rename val -> newValue

karalabe committed:
Merge pull request #22573 from holiman/checkfreq

core: fix condition on header verification

MariusVanDerWijden committed:
eth: dump rpc gas cap and tx fee cap (#22574)

karalabe committed:
eth/protocols, metrics, p2p: add handler performance metrics

karalabe committed:
Merge pull request #22581 from karalabe/network-latency-monitoring

eth/protocols, metrics, p2p: add handler performance metrics

karalabe committed:
eth/protocols, metrics: use resetting histograms for rare packets

karalabe committed:
Merge pull request #22586 from karalabe/resettting-histogram

eth/protocols, metrics: use resetting histograms for rare packets

zouguangxian committed:
eth: fix corner case in sync head determination (#21695)

This avoids synchronisation failures when the local header is ahead of
the local full block.

fjl committed:
cmd/geth, consensus/ethash: add support for --miner.notify.full flag (#22558)

The PR implements the --miner.notify.full flag that enables full pending block
notifications. When this flag is used, the block notifications sent to mining
endpoints contain the complete block header JSON instead of a work package
array.

Co-authored-by: AlexSSD7 <alexandersadovskyi7@protonmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
metrics/influxdb: don't push empty histograms, no measurement != 0

karalabe committed:
Merge pull request #22590 from karalabe/nil-histogram

metrics/influxdb: don't push empty histograms, no measurement != 0

karalabe committed:
eth/protocols/snap: add peer id and req id to the timeout logs

karalabe committed:
Merge pull request #22591 from karalabe/snap-detailed-timeout-logs

eth/protocols/snap: add peer id and req id to the timeout logs

fjl committed:
cmd/devp2p: update to newer cloudflare API client (#22588)

This upgrades the cloudflare client dependency to v0.14.0. The new
version changes the API because all methods now require a context
parameter. This change also reduces the log level of the 'Skipping...'
message to debug, following a similar change in the AWS deployer.

rjl493456442 committed:
core/state/pruner: move the compaction out of the pruning procedure (#22579)

The main idea behind it is: the range compaction is very expensive
which can take a few hours to finish. During this long procedure,
a lot of exceptions can occur, e.g.
- Geth is killed manually
- Geth is killed because of machine crash
- etc

In order to minimize the effect of the exceptions, the compaction
is moved out of the pruning. So that even the compaction is not
finished, the pruning is regarded as done.

holiman committed:
eth/protocols/snap: try to prevent requests timing out

renaynay committed:
core: add BlockGen.GetBalance method  (#22589)

MariusVanDerWijden committed:
cmd/puppeth: specify working directory for nodejs 15 (#22549)

karalabe committed:
Merge pull request #22596 from holiman/snap_timelimits

eth/protocols/snap: try to prevent requests timing out

holiman committed:
cmd/geth: add db dumptrie command (#22563)

Adds the command "geth db dumptrie <root> <seek> <max>", to better help investigate the trie data

nebojsa94 committed:
core/vm: fix Byzantium address list (#22603)

MariusVanDerWijden committed:
ethstats: avoid creating subscriptions on background goroutine (#22587)

This fixes an issue where the ethstats service could crash if geth was
started and then immediately stopped due to an internal error. The
cause of the crash was a nil subscription being returned by the backend,
because the background goroutine creating them was scheduled after
the backend had already shut down.

Moving the creation of subscriptions into the Start method, which runs
synchronously during startup of the node, means the returned subscriptions
can never be 'nil'.

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
core/state/snapshot, ethdb: track deletions more accurately (#22582)

* core/state/snapshot, ethdb: track deletions more accurately

* core/state/snapshot: don't reset the iterator, leveldb's screwy

* ethdb: don't mess with the insert batches for now

fjl committed:
rpc: tighter shutdown synchronization in client subscription (#22597)

This fixes a rare issue where the client subscription forwarding loop
would attempt send on the subscription's channel after Unsubscribe has
returned, leading to a panic if the subscription channel was already
closed by the user. Example:

    sub, _ := client.Subscribe(..., channel, ...)
    sub.Unsubscribe()
    close(channel)

The race occurred because Unsubscribe called quitWithServer to tell the
forwarding loop to stop sending on sub.channel, but did not wait for the
loop to actually come down. This is fixed by adding an additional channel
to track the shutdown, on which Unsubscribe now waits.

Fixes #22322

karalabe committed:
all: fix miner hashRate -> hashrate on API calls

karalabe committed:
Merge pull request #22604 from karalabe/hashrate-capitalization-fix

all: fix miner hashRate -> hashrate on API calls

fxfactorial committed:
core/state/snapshot: fix data race in diff layer (#22540)

piersy committed:
internal/ethapi: fix eth_chainId method (#22243)

This removes the duplicated definition of eth_chainID
in package eth and updates the definition in internal/ethapi
to treat chain ID as a bigint.

Co-authored-by: Felix Lange <fjl@twurst.com>

AmitBRD committed:
graphql: add support for tx types and tx access lists (#22491)

This adds support for EIP-2718 access list transactions in the GraphQL API.

Co-authored-by: Amit Shah <amitshah0t7@gmail.com>
Co-authored-by: Felix Lange <fjl@twurst.com>

s1na committed:
internal/debug: add JSON log format and rename logging flags (#22341)

This change adds support for logging JSON records when the --log.json flag is
given. The --debug and --backtrace flags are deprecated and replaced by
--log.debug and --log.backtrace.

While changing this, it was noticed that the --memprofilerate and
--blockprofilerate were ineffective (they were always overridden even if
--pprof.memprofilerate was not set). This is also fixed.

Co-authored-by: Felix Lange <fjl@twurst.com>

petesimard committed:
cmd/utils: move cache sanity check to SetEthConfig (#22510)

Move the cache sanity check to the SetEthConfig function
to allow the config file to load.

Evolution404 committed:
consensus/ethash: replace a magic number with it's constant (#22618)

zsfelfoldi committed:
les: move client pool to les/vflux/server (#22495)

* les: move client pool to les/vflux/server

* les/vflux/server: un-expose NodeBalance, remove unused fn, fix bugs

* tests/fuzzers/vflux: add ClientPool fuzzer

* les/vflux/server: fixed balance tests

* les: rebase fix

* les/vflux/server: fixed more bugs

* les/vflux/server: unexported NodeStateMachine fields and flags

* les/vflux/server: unexport all internal components and functions

* les/vflux/server: fixed priorityPool test

* les/vflux/server: polish balance

* les/vflux/server: fixed mutex locking error

* les/vflux/server: priorityPool bug fixed

* common/prque: make Prque wrap-around priority handling optional

* les/vflux/server: rename funcs, small optimizations

* les/vflux/server: fixed timeUntil

* les/vflux/server: separated balance.posValue and negValue

* les/vflux/server: polish setup

* les/vflux/server: enforce capacity curve monotonicity

* les/vflux/server: simplified requestCapacity

* les/vflux/server: requestCapacity with target range, no iterations in SetCapacity

* les/vflux/server: minor changes

* les/vflux/server: moved default factors to balanceTracker

* les/vflux/server: set inactiveFlag in priorityPool

* les/vflux/server: moved related metrics to vfs package

* les/vflux/client: make priorityPool temp state logic cleaner

* les/vflux/server: changed log.Crit to log.Error

* add vflux fuzzer to oss-fuzz

Co-authored-by: rjl493456442 <garyrong0905@gmail.com>

rjl493456442 committed:
eth, les: fix tracers (#22473)

* eth, les: fix tracer

* eth: isolate live trie database in tracer

* eth: fix nil

* eth: fix

* eth, les: add checkLive param

* eth/tracer: fix

MariusVanDerWijden committed:
core, eth, internal/ethapi: create access list RPC API (#22550)

* core/vm: implement AccessListTracer

* eth: implement debug.createAccessList

* core/vm: fixed nil panics in accessListTracer

* eth: better error messages for createAccessList

* eth: some fixes on CreateAccessList

* eth: allow for provided accesslists

* eth: pass accesslist by value

* eth: remove created acocunt from accesslist

* core/vm: simplify access list tracer

* core/vm: unexport accessListTracer

* eth: return best guess if al iteration times out

* eth: return best guess if al iteration times out

* core: docstring, unexport methods

* eth: typo

* internal/ethapi: move createAccessList to eth package

* internal/ethapi: remove reexec from createAccessList

* internal/ethapi: break if al is equal to last run, not if gas is equal

* internal/web3ext: fixed arguments

* core/types: fixed equality check for accesslist

* core/types: no hardcoded vals

* core, internal: simplify access list generation, make it precise

* core/vm: fix typo

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
eth: fix tracing state retrieval if requesting the non-dirty genesis

karalabe committed:
Merge pull request #22629 from karalabe/fix-chain-tracing-from-0

eth: fix tracing state retrieval if requesting the non-dirty genesis

meowsbits committed:
build: use env-aware bash shebangs for evmc-build scripts

Date: 2021-04-07 11:16:10-05:00
Signed-off-by: meows <b5c6@protonmail.com>

karalabe committed:
params: update CHTs for v1.10.2

karalabe committed:
Merge pull request #22632 from karalabe/cht-1.10.2

params: update CHTs for v1.10.2

fjl committed:
params: release go-ethereum v1.10.2 stable

meowsbits committed:
Merge remote-tracking branch 'etclabscore/merge/foundation-release/1.10.1-resolved-stateDiff' into merge/foundation-release/1.10.1-resolved

meowsbits committed:
Merge branch 'foundation-1.10.2' into merge/foundation-release/1.10.2-resolved

 Conflicts:
	cmd/devp2p/internal/ethtest/suite.go
	cmd/devp2p/internal/ethtest/types.go
	cmd/geth/chaincmd.go
	cmd/geth/main.go
	cmd/puppeth/module_dashboard.go
	cmd/utils/flags.go
	consensus/ethash/algorithm.go
	consensus/ethash/algorithm_test.go
	consensus/ethash/ethash.go
	consensus/ethash/sealer_test.go
	core/rawdb/database.go
	core/rawdb/freezer.go
	core/state_transition.go
	core/tx_pool.go
	core/vm/contracts.go
	core/vm/eips.go
	core/vm/evm.go
	core/vm/instructions.go
	core/vm/logger.go
	core/vm/logger_json.go
	core/vm/runtime/runtime.go
	core/vm/runtime/runtime_test.go
	eth/api.go
	eth/backend.go
	eth/ethconfig/config.go
	eth/state_accessor.go
	eth/tracers/api.go
	eth/tracers/tracer.go
	go.mod
	go.sum
	internal/ethapi/api.go
	les/peer.go
	les/server.go
	params/config.go
	params/version.go

meowsbits committed:
go.mod,go.sum: go get -d ./...

Date: 2021-04-13 10:14:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2021-04-13 10:15:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/rawdb: fixup readonly vals for FreezerRemote

Date: 2021-04-13 10:17:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/ethapi: fixup precompiles list construction for config/blocknum

Date: 2021-04-13 10:23:24-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: implement missing tracer interface method CapturePreEVM

Date: 2021-04-13 10:23:45-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: remove disused ActivePrecompiles method

We already can build a superset value of this
via PrecompiledContractsForConfig.

Date: 2021-04-13 10:24:57-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/puppeth: rm -rf cmd/puppeth

Upstream added a new dashboard html file.

We have purged puppeth.

Date: 2021-04-13 10:26:19-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils,node: fixup node database freezer remote readonly signature

Date: 2021-04-13 10:27:35-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: fixup ethash config construction sytax for test

Date: 2021-04-13 10:29:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/vm, eth/tracers: fix conflicts with Tracer interface

ziogaschr committed:
eth/tracers: fix tracer related conflicts

ziogaschr committed:
eth/ethconfig: fix failed/broken merge

ziogaschr committed:
cmd/echaindb: conform to NewLevelDBDatabase readonly mode

ziogaschr committed:
core/vm, eth/tracers: fix lint issues on tracers

ziogaschr committed:
core: fix remote freezer tests for readonly Database

meowsbits committed:
graphql: fixup core.Genesis -> genesisT.Genesis imports

Date: 2021-04-13 15:55:51-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
graphql: type ChainConfigurator has no field ChainID

Date: 2021-04-13 15:56:12-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: [PTAL] remove have/want from error string

I haven't figured out where this is getting
interpolated to the error value.

It seems the error is 'right', but is
not being formatted with the details.

Date: 2021-04-13 17:02:32-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits commented on core/vm/instructions.go:
```diff
@@ -543,42 +543,43 @@ func opJumpi(pc *uint64, interpreter *EVMInterpreter, callContext *callCtx) ([]b
 	// reuse size int for stackvalue
 	stackvalue := size
 
+	scope.Contract.UseGas(gas)
```
:eyes:

ziogaschr replied:
Great catch. The merge brought in old (touched upstream) stuff that we moved to `gas_table.go`, because of this we were removing the gas used twice.

Resolved by b941602

meowsbits commented on go.mod:
```diff
@@ -31,46 +35,51 @@ require (
 	github.com/status-im/keycard-go v0.0.0-20200402102358-957c09536969
 	github.com/stretchr/testify v1.7.0
-	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca
+	github.com/syndtr/goleveldb v1.0.1-0.20210305035536-64b5b1c73954
```
:eyes:

meowsbits commented on cmd/geth/dbcmd.go:
```diff
@@ -59,52 +59,124 @@ Remove blockchain and state databases`,
+		Flags: []cli.Flag{
+			utils.DataDirFlag,
+			utils.SyncModeFlag,
+			utils.MainnetFlag,
```
- [x] Add our additional chain flags.

meowsbits replied:
Done with https://github.com/etclabscore/core-geth/pull/357/commits/49bd5e70385382528187bd42365392b4a91fad5f

meowsbits commented on cmd/geth/chaincmd.go:
```diff
@@ -66,7 +62,11 @@ It expects the genesis file as argument.`,
 		ArgsUsage: "",
 		Flags: []cli.Flag{
-			utils.DataDirFlag,
+			utils.MainnetFlag,
```
- [x] Add our additional chain flags.

meowsbits replied:
49bd5e70385382528187bd42365392b4a91fad5f

meowsbits commented on cmd/echaindb/cmd/readChainconfig.go:
```diff
@@ -47,7 +47,7 @@ Example:
 
 		log.Println("Opening database...")
-		db, err := rawdb.NewLevelDBDatabase(chainDBPath, 256, 16, "")
+		db, err := rawdb.NewLevelDBDatabase(chainDBPath, 256, 16, "", true)
```
We should maybe think about getting rid of this executable (cmd/echaindb) since its functionality is being overtaken by that of `geth`'s `db` command (and never was developed beyond what I happened to need it for once upon a time).

meowsbits replied:
Proposed with https://github.com/etclabscore/core-geth/pull/366

ziogaschr committed:
ethclient: fix TestRPCDiscover test

ziogaschr committed:
core/vm: apply CREATE|CREATE2 instruction gas usage logic

…broken because of the upstream merge

ziogaschr committed:
eth/tracers: continue using returnData in JS tracer CaptureState

It is needed by the callTracerParity for maintaining 100% OE compatibility

ziogaschr committed:
eth/tracers: call_tracer_parity to use the blockNumber passed in result

ziogaschr committed:
eth/tracers: use txTraceContext instead of taskExtraContext where possible

meowsbits committed:
core: simplify header chain nil check

Date: 2021-04-28 06:15:26-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
internal/debug: fix dropped deprecated (legacy) pprof flags

All depcrecated pprof flags were dropped, apparently
accidentally.

I noticed this because --debug was unavailable.

Not only were the flag declarations unregistered,
but logical support for pprof.port, addr, and cpuprofile
were missing; this adds support for them.

Date: 2021-04-28 06:33:15-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/geth: add classic,mordor,kotti flags for dbcmd and chaincmd

Date: 2021-04-28 06:52:00-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
build,cmd/echaindb: rm -rf and uses of echaindb command

This small program's functionality has been
replace with the new 'geth db' command.

Date: 2021-04-28 07:00:28-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #366 from etclabscore/merge/foundation-release/1.10.2-resolved-rmrf-cmd-echaindb

build,cmd/echaindb: rm -rf and uses of echaindb command

ziogaschr reviewed (approved):
LGTM to be merged at v1.10.1

meowsbits committed:
Merge branch 'merge/foundation-release/1.10.1-resolved' into merge/foundation-release/1.10.2-resolved
